### ISSUE TITLE
Warning message about non unique database keys

### ISSUE THREAD

**Issue by pepper-jk** [tklab-tud/ID2T#42]
**Issue by [carlos.garcia](https://git.tk.informatik.tu-darmstadt.de/carlos.garcia)**
_Friday May 11, 2018 at 16:37 GMT_
_Originally opened as https://git.tk.informatik.tu-darmstadt.de/SPIN/ID2T-toolkit/issues/117_

----

When analyzing the file found in [1], the following error is displayed when writing the statistics to the database:
```
Exception in statistics_db: UNIQUE constraint failed: ip_ports.ipAddress, ip_ports.portDirection, ip_ports.portNumber
```

[1]: http://www.fukuda-lab.org/mawilab/v1.1/2018/04/04/20180404.html

**Comment by pepper-jk** [comment]
**Comment by [carlos.garcia](https://git.tk.informatik.tu-darmstadt.de/api/v1/repos/SPIN/ID2T-toolkit/carlos.garcia)**
_Friday May 11, 2018 at 17:14 GMT_

----

This warning was more serious than expected: the `ip_ports` table would not be created.

The issue happens when IP A sends packets to B using the same ports but different protocols (i.e., UDP and TCP).

Solution: change the primary key of `ip_ports` to include the protocol
```sql
PRIMARY KEY(ipAddress,portDirection,portNumber,portProtocol)
```

**Comment by pepper-jk** [comment]
**Comment by [jens.klein](https://git.tk.informatik.tu-darmstadt.de/api/v1/repos/SPIN/ID2T-toolkit/jens.klein)**
_Tuesday May 22, 2018 at 20:17 GMT_

----

**Comment by pepper-jk** [comment]
**Comment by [jens.klein](https://git.tk.informatik.tu-darmstadt.de/api/v1/repos/SPIN/ID2T-toolkit/jens.klein)**
_Tuesday May 22, 2018 at 20:19 GMT_

----

Was fixed in pull request #120.
